src/main/scala/mimir/adaptive/SchemaMatching.scala

```diff
@@ -0,0 +1,169 @@
+          (
+            varName.toString.toUpperCase -> 
+              Type.fromString(typeName.toString)
+          )
+        }).
+        toList
+
+    val modelsByType = 
+      ModelRegistry.schemamatchers.toSeq.map {
+        case (
+          modelCategory:String, 
+          constructor:ModelRegistry.SchemaMatchConstructor
+        ) => {
+          val modelsByColAndType =
+            constructor(
+              db, 
+              s"$viewName:$modelCategory", 
+              Left(config.query), 
+              Right(targetSchema)
+            ).toSeq.map {
+              case (col, (model, idx)) => (col, (model, idx, Seq[Expression]()))
+            }
+
+          (modelCategory, modelsByColAndType)
+        }
+      }
+
+    val (
+      candidateModels: Map[String,Seq[(String,Int,Seq[Expression],String)]],
+      modelEntities: Seq[Model]
+    ) = 
+      LensUtils.extractModelsByColumn(modelsByType)
+
+    // Sanity check...
+    targetSchema.map(_._1).foreach( target => {
+      if(!candidateModels.contains(target)){ 
+        throw new Exception("No valid schema matching model for column '"+target+"' in lens '"+viewName+"'");
+      }
+    })
+
+
+    val (
+      schemaChoice:List[(String,Expression)], 
+      metaModels:List[Model]
+    ) =
+      candidateModels.
+        toList.
+        map({ 
+          case (column, models) => {
+            //TODO: Replace Default Model
+            val metaModel = new DefaultMetaModel(
+                s"$viewName:META:$column", 
+                s"picking a source for column '$column'",
+                models.map(_._4)
+              )
+            val metaExpr = LensUtils.buildMetaModel(
+              metaModel.name, 0, Seq[Expression](), Seq[Expression](),
+              models, Seq[Expression]()
+            )
+
+            ( (column, metaExpr), metaModel )
+          }
+        }).
+        unzip
+
+    val proxyModel = SchemaMatchingProxyModel(s"$viewName:PROXY:SCHEMA_MATCHING", targetSchema, targetSchema.map(sche => modelEntities(modelEntities.indexWhere(_.name.endsWith(s":${sche._1}")) ).name ))   
+    modelEntities ++ metaModels :+ proxyModel
+  }
+
+  def tableCatalogFor(db: Database, config: MultilensConfig): Operator =
+  {
+    HardTable(
+      Seq(
+        ("TABLE_NAME",TString())
+      ),
+      Seq(
+        Seq(
+          StringPrimitive("DATA")
+        )
+      )
+    )
+  }
+  
+  def attrCatalogFor(db: Database, config: MultilensConfig): Operator =
+  {
+    HardTable(
+      Seq(
+        ("TABLE_NAME" , TString()), 
+        ("ATTR_TYPE", TType()),
+        ("IS_KEY", TBool()),
+        ("COLUMN_ID", TInt())
+      ),
+      (0 until config.args.size).map { col =>
+        Seq(
+          StringPrimitive("DATA"),
+          TypePrimitive(TString()),
+          BoolPrimitive(false),
+          IntPrimitive(col)
+        )
+      }
+    ).addColumn(
+      "ATTR_NAME" -> VGTerm(s"${config.schema}:PROXY:SCHEMA_MATCHING" , 1, Seq(Var("COLUMN_ID")), Seq())
```

Comment:
Why make the attribute name uncertain?

Reply:
I realize that the ATTR_NAME is not actually uncertain, but I'm not sure how else to the encode uncertainty of the schema matching source schema mapping at the schema level.  Suggestions?

Reply:
As per live discussion, the way to go here should be: 
```
SELECT[VGTerm(MatchModel, Target.Name) = Source.Name](
  JOIN(
    HardTable( ... target schema ...)
    HardTable( ... source schema ...)
  )
)
```